### PR TITLE
[CWS] add access to envs value through SECL

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This allows to write rules like

```
exec.envp =~ "LD_LIBRAY_PATH=/tmp*"
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by alai97 — APPROVED** [review]
Looks good for docs!

**Review by paulcacheux — APPROVED** [review]

File: pkg/security/secl/model/process_cache_entry.go [hunk 1]
```diff
@@ -210,30 +210,21 @@ func (p *ArgsEntry) ToArray() ([]string, bool) {
...
 	parsed bool
 	keys   []string
+	kv     map[string]string
 }
 
-// ToMap returns envs as map
-func (p *EnvsEntry) ToMap() (map[string]string, bool) {
-	if p.Values != nil || p.parsed {
+// ToArray returns envs as map
+func (p *EnvsEntry) ToArray() ([]string, bool) {
```
Comment (lebauce) [inline pkg/security/secl/model/process_cache_entry.go:222]: not: update comment

**Review by lebauce — APPROVED** [review]

File: pkg/security/secl/model/process_cache_entry.go [hunk 2]
```diff
@@ -251,25 +242,41 @@ func (p *EnvsEntry) Keys() ([]string, bool) {
 		return p.keys, p.Truncated
 	}
 
-	if !p.parsed {
-		p.ToMap()
-	}
-
-	p.keys = make([]string, len(p.Values))
+	values, _ := p.ToArray()
```
Comment (lebauce) [inline pkg/security/secl/model/process_cache_entry.go:245]: nit:
if len(values) == 0 {
    return nil, p.Truncated
}

Comment (lebauce) [inline pkg/security/secl/model/process_cache_entry.go:245]: if p.keys != nil {
    return p.keys, p.Truncated
}

values, _ := p.ToArray()
if len(values) == 0 {
    return nil, p.Truncated
}

**Review by lebauce — APPROVED** [review]

**Review by lebauce — APPROVED** [review]

**Pull request merged**